homeassistant/helpers/intent.py

```diff
@@ -419,20 +419,30 @@ async def async_handle_states(
         service_coros = []
         for state in states:
             service_coros.append(self.async_call_service(intent_obj, state))
-            success_results.append(
-                IntentResponseTarget(
-                    type=IntentResponseTargetType.ENTITY,
-                    name=state.name,
-                    id=state.entity_id,
-                ),
-            )
 
-        # Handle service calls in parallel.
-        # We will need to handle partial failures here.
-        await asyncio.gather(*service_coros)
+        # Handle service calls in parallel, noting failures as they occur.
+        failed_results: list[IntentResponseTarget] = []
+        for service_coro in asyncio.as_completed(service_coros):
```

Comment:
since you're still waiting for all of them, what is the benefit of using `as_completed` here?

Reply:
It just let me combine the `await` and `for` loop to add the response targets into a single call, vs. a `wait` followed by a loop. As I understand, they're still run concurrently.